core/tauri/src/resources/plugin.rs

```diff
@@ -0,0 +1,22 @@
+// Copyright 2019-2023 Tauri Programme within The Commons Conservancy
+// SPDX-License-Identifier: Apache-2.0
+// SPDX-License-Identifier: MIT
+
+use crate::{
+  command,
+  plugin::{Builder, TauriPlugin},
+  AppHandle, Runtime,
+};
+
+use super::ResourceId;
+
+#[command(root = "crate")]
+fn close<R: Runtime>(app: AppHandle<R>, rid: ResourceId) -> crate::Result<()> {
```

Comment:
this is problematic, an attacker can close all resources by iterating on IDs until an error is returned.

maybe we should make the ID random instead?

Reply:
Closing resource is not that bad right?

Also for reference, deno is using the same ID strategy and though they also provide `.close()` you can't really invoke the underlying mechanism of it (compared to `invoke("plugin:resources|close"`). 

So in order to simplify the implementation and keep using incrementing ID numbers, we can hide the invoke endpoint and randomize it on each launch of the app.